### PR TITLE
fix: handle boundary conditions in vmv.v.i operation

### LINKED ISSUE #1618: Focus test for vmv_v_i still asserts on ISS but passes on Sail now
@ishaan-arora-1 fixed issue #1599 by correcting the hex arguments to the `vmv_v_i` instructions in [`vmv_v_i.S`](https://github.com/riscv/riscv-unified-db/blob/main/tests/isa/rv64uv/vmv_v_i.S).

After recompilation the test now passes on Sail, but the ISS still gives the same assertion error. 

@jmawet Please note (and skip the `vmv_v_i` test for your development. You may want to consult with @henrikg-qc  re the ISS assertion.

- Sail (Binary Release 0.9): Snippet for `test_1`
>$ sail_riscv_sim --trace-instr --config /opt/sail-riscv/share/sail-riscv/config/rv64d_v128_e64.json tests/isa/rv64uv-p-vmv_v_i
...
[77] [U]: 0x00000000800001A4 (0x00100193) addi x3, x0, 0x1                          test_1+0
[78] [U]: 0x00000000800001A8 (0xC1817057) vsetivli x0, 0x2, e64, m1, tu, mu         test_1+4
[79] [U]: 0x00000000800001AC (0x5E02B0D7) vmv.v.i v1, 0x5                           test_1+8
[80] [U]: 0x00000000800001B0 (0x421022D7) vmv.x.s x5, v1                            test_1+12
[81] [U]: 0x00000000800001B4 (0x00500393) addi x7, x0, 0x5                          test_1+16
[82] [U]: 0x00000000800001B8 (0x06729463) bne x5, x7, 0x68                          test_1+20
...
SUCCESS

- ISS
> $ ./do test:riscv_vector_tests CONFIG=rv64 BUILD_TYPE=DEBUG IGNOREUNDEFINED=YES JOBS=8
...
/home/adingank/git/forks/riscv-unified-db/gen/cpp_hart_gen/rv64_Debug/build/iss -m rv64 -c /home/adingank/git/forks/riscv-unified-db/cfgs/rv64-vector.yaml tests/isa/rv64uv-p-vmv_v_i
...
PC 800001a0 mret
PC 800001a4 addi x3, x0, 1
R x0 0
R= x3 1
PC 800001a8 vsetivli x0, 2, 24
R= x0 0
PC 800001ac vmv.v.i 1, 5
At /home/adingank/git/forks/riscv-unified-db/gen/cpp_hart_gen/rv64_Debug/include/udb/bits.hpp:2814:7: constexpr udb::_PossiblyUnknownRuntimeBits<constmax_v<MsbN, LsbN>, false> udb::_PossiblyUnknownBits<N, Signed>::extract(const MsbType<MsbN, false>&, const LsbType<LsbN, false>&) const [with MsbType = udb::_Bits; unsigned int MsbN = 128; LsbType = udb::_Bits; unsigned int LsbN = 32; unsigned int N = 128; bool Signed = false] :
   Assertion failed: (msb >= lsb)
   Negative range is not allowed
/usr/lib/ruby/gems/3.4.0/gems/rake-13.3.1/lib/rake/file_utils.rb:65:in 'block in FileUtils#create_shell_runner': Command failed with status (): [/home/adingank/git/forks/riscv-unified-db/gen/cpp_hart_gen/rv64_Debug/build/iss -m rv64 -c /home/adingank/git/forks/riscv-unified-db/cfgs/rv64-vector.yaml tests/isa/rv64uv-p-vmv_v_i] (RuntimeError)
        from /usr/lib/ruby/gems/3.4.0/gems/rake-13.3.1/lib/rake/file_utils.rb:57:in 'FileUtils#sh'
        from /home/adingank/git/forks/riscv-unified-db/backends/cpp_hart_gen/tasks.rake:521:in 'block (3 levels) in <top (required)>'
        from /home/adingank/git/forks/riscv-unified-db/backends/cpp_hart_gen/tasks.rake:520:in 'Array#each'
        from /home/adingank/git/forks/riscv-unified-db/backends/cpp_hart_gen/tasks.rake:520:in 'block (2 levels) in <top (required)>'
        from /usr/lib/ruby/gems/3.4.0/gems/rake-13.3.1/lib/rake/task.rb:281:in 'block in Rake::Task#execute'
        from /usr/lib/ruby/gems/3.4.0/gems/rake-13.3.1/lib/rake/task.rb:281:in 'Array#each'
        from /usr/lib/ruby/gems/3.4.0/gems/rake-13.3.1/lib/rake/task.rb:281:in 'Rake::Task#execute'
        from /usr/lib/ruby/gems/3.4.0/gems/rake-13.3.1/lib/rake/task.rb:219:in 'block in Rake::Task#invoke_with_call_chain'
        from /usr/lib/ruby/gems/3.4.0/gems/rake-13.3.1/lib/rake/task.rb:199:in 'Monitor#synchronize'
        from /usr/lib/ruby/gems/3.4.0/gems/rake-13.3.1/lib/rake/task.rb:199:in 'Rake::Task#invoke_with_call_chain'
        from /usr/lib/ruby/gems/3.4.0/gems/rake-13.3.1/lib/rake/task.rb:188:in 'Rake::Task#invoke'
        from /usr/lib/ruby/gems/3.4.0/gems/rake-13.3.1/lib/rake/application.rb:188:in 'Rake::Application#invoke_task'
        from /usr/lib/ruby/gems/3.4.0/gems/rake-13.3.1/lib/rake/application.rb:138:in 'block (2 levels) in Rake::Application#top_level'
        from /usr/lib/ruby/gems/3.4.0/gems/rake-13.3.1/lib/rake/application.rb:138:in 'Array#each'
        from /usr/lib/ruby/gems/3.4.0/gems/rake-13.3.1/lib/rake/application.rb:138:in 'block in Rake::Application#top_level'
        from /usr/lib/ruby/gems/3.4.0/gems/rake-13.3.1/lib/rake/application.rb:147:in 'Rake::Application#run_with_threads'
        from /usr/lib/ruby/gems/3.4.0/gems/rake-13.3.1/lib/rake/application.rb:132:in 'Rake::Application#top_level'
        from -e:1:in `'<main>'`

### PR DESCRIPTION
**Pull request by lucifer4330k**
The vmv.v.i instruction implementation previously used unconditional vector slicing to update the destination register, which caused assertion failures in the ISS when accessing invalid bit ranges (e.g., msb < lsb). This commit introduces conditional logic to handle cases where the modified element is at the start (bit 0) or end (bit VLEN-1) of the vector register, ensuring only valid slices are concatenated. Fixes issue with vmv.v.i causing Assertion failed: (msb >= lsb) in ISS.
fixes #1618

### CODE REVIEW

File: spec/std/isa/inst/V/vmv.v.i.yaml [hunk 1]
```diff
@@ -55,11 +55,27 @@ operation(): |
...
+        sext_imm[state.sew-1:0],
+        v[vd][start_bit_pos-1:0]
+      };
+    } else {
+      v[vd] = {
+        v[vd][VLEN-1:end_bit_pos + 1],
+        sext_imm[state.sew-1:0],
+        v[vd][start_bit_pos-1:0]
+      };
+    }
```
Comment (ThinkOpenly) [inline spec/std/isa/inst/V/vmv.v.i.yaml:78]: To my naive eyes, this looks overly complicated. It seems like a fairly simple assignment would suffice:
```
v[vd][start_bit_pos:end_bit_pos] = sext_imm[state.sew-1:0]
```
(Am I missing something?)
